storage/remote/write_handler.go

```diff
@@ -533,24 +533,27 @@ type OTLPOptions struct {
 	// LookbackDelta is the query lookback delta.
 	// Used to calculate the target_info sample timestamp interval.
 	LookbackDelta time.Duration
+	// IngestCTZeroSample enables writing zero samples based on the start time
+	// of metrics.
+	IngestCTZeroSample bool
 }
 
 // NewOTLPWriteHandler creates a http.Handler that accepts OTLP write requests and
 // writes them to the provided appendable.
-func NewOTLPWriteHandler(logger *slog.Logger, _ prometheus.Registerer, appendable storage.Appendable, configFunc func() config.Config, opts OTLPOptions) http.Handler {
+func NewOTLPWriteHandler(logger *slog.Logger, reg prometheus.Registerer, appendable storage.Appendable, configFunc func() config.Config, opts OTLPOptions) http.Handler {
```

Comment:
This should take an `appendable CombinedAppender` interface , not `storage.Appendable` so we can seed it with a different implementation. 

Of course it would mean that in api.go `NewAPI` should also take it as an input parameter (dependency inject) so we can implement in Mimir and not in mimir-prometheus - removing the deviation from upstream completely.

?

Reply:
What public function does mimir use?  NewAPI?  Or NewOTLPWriteHandler?

Reply:
Discussed in the SIG meeting.  They use NewAPI.

Reply:
Hold on for now, I might be able to use `NewPrometheusConverter` directly. Working on it.